### PR TITLE
Issue #15026: Added support to check input file against whole google configuration for google IT tests

### PR DESCRIPTION
**Pull request by Zopsss**
issue #15026 

created a new method: `verifyWithWholeGoogleConfig()` which helps us to check the input file against whole google configuration. We just require to pass the file path to the method and everything else will work like a magic 🔮🪄

### CODE REVIEW

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java [hunk 1]
```diff
@@ -500,6 +500,28 @@ public final void verifyWithGoogleConfigParser(String[] listOfModules,
...
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
+        final Configuration googleConfig = getMasterGoogleConfig();
```
Comment (rnveach) [inline src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:510]: This type of method should not be in `AbstractIt`. This class is used for multiple different areas of IT, including Sun and CS's regression area. It doesn't make sense for sun to be able to see google's config. This is why the abtsracts are separated out like this. This type of mixing is almost like you gave production code access to test code. They should be kept separate.

Why can't we have:
* `verifyWithItConfig` method in this abstract class, which will take in the entire config.
* `verifyWithWholeConfig` in `AbstractGoogleModuleTestSupport` which will call `verifyWithItConfig` and pass it Google's config.

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java [hunk 2]
```diff
@@ -491,7 +491,7 @@ protected final void verify(Checker checker,
      */
     public final void verifyWithGoogleConfigParser(String[] listOfModules,
```
Comment (rnveach) [inline src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:492]: Same here.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java (hunk 2 above)
Comment (rnveach) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:492]: Same with `getMasterGoogleConfig`, `getGoogleConfigChildren`, and anything else that says "google" in this abstract IT class.

Comment (rnveach) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:492]: It looks like some methods could just be renamed and not split apart.

`getGoogleConfigChildren` really has no code specific to google, except in method name and one parameter name. So it can be renamed.

`getMasterGoogleConfig` looks like it could be made abstract and renamed to `getItConfig`, and the method in `AbstractGoogleModuleTestSupport` could just return `CONFIGURATION` which it already loads and caches for you.

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java (hunk 1 above)
Comment (romani) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:510]: Agree, nothing special for Google here. It just execution based on some config. Each exact test will define what config to use.

Comment (Zopsss) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:510]: so we will need to make changes for `verifyWithGoogleConfigParser()` too to meet these criteria.

Comment (rnveach) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:510]: >Each exact test will define what config to use.

`AbstractGoogle` should define the config to use. Test should define the verify method to use.

Comment (romani) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java:510]: Yes, we can have some abstract class for all Google ITs to define path it.
Parent class for `abstract class for all Google ITs` should just receive any path to config and just load it.

Comment (Zopsss) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java]: I have moved all the google related methods to the `AbstractGoogleModuleTestSupport` and kept only required methods in `AbstractItModuleTestSupport`

created a new [verifyWithItConfig()](https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L465) as @rnveach suggested.

This method takes a configuration and file path and performs the verification. We're not loading configuration in this method, we're receiving it from parameters, here is why I think it is better:

In future if we decide to migrate sun style to chapter wise testing or add new style which will follow chapter wise testing, it can work in the same way as google. We will load it's configuration from it's abstract class and pass it to this method as we're doing currently ( providing config from `AbstractGoogleModuleTestSupport` ). This will remove the overhead of loading all types of configuration from one place. We can provide the configuration of different styles from their corresponding abstract class. This is more flexible way.

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java (hunk 2 above)
Comment (Zopsss) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java]: > getGoogleConfigChildren really has no code specific to google, except in method name and one parameter name. So it can be renamed.

This method is only used inside [verifyWithGoogleConfigParser()](https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L163C23-L163C51) for creating the google config. So instead of renaming it, I have moved it to `AbstractGoogleModuleTestSupport` with other methods.

> getMasterGoogleConfig looks like it could be made abstract and renamed to getItConfig, and the method in AbstractGoogleModuleTestSupport could just return CONFIGURATION which it already loads and caches for you.

In the new changes, I created the `verifyWithItConfig()` method which will directly receive the configuration from `AbstractGoogleModuleTestSupport` as its parameter. So we will not need `getItConfig` to get the configuration.

Comment (Zopsss) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java]: Here's the new code for you to understand it better:

https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L138-L210

https://github.com/checkstyle/checkstyle/blob/c3f84d08cce5ad488d54a0ca210bfcb0b475bb69/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L458-L471

also take a look at this comment https://github.com/checkstyle/checkstyle/pull/15028#discussion_r1646095246, here I have explained the reason behind why I kept working of `verifyWithItConfig` like this.

File: src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java (hunk 1 above)
Comment (Zopsss) [reply src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java]: Now the `AbstractItModuleTestSupport` has only `verifyWithItConfig()` and some other helper methods to perform the verification. All the google related methods has been moved to `AbstractGoogleModuleTestSupport`

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 3]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+    }
+
+    /**
+     * Gets the specified children module(s) from google config.
+     *
+     * @param module the children module to get. It can be either a module name or module id.
+     * @return list of children modules.
+     * @throws CheckstyleException if there is a problem for getting the module.
+     */
+    private static List<Configuration> getGoogleConfigChildren(String module)
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: `getConfigChildren`

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 4]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+
+        if (treeWalkerConfig.getChildren().length > 0) {
+            googleConfig.addChild(treeWalkerConfig);
+        }
+
+        verifyWithItConfig(googleConfig, filePath);
+    }
+
+    /**
+     * Gets the specified children module(s) from google config.
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:192]: remove "google"

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java [hunk 5]
```diff
@@ -1,35 +1,44 @@
 package com.google.checkstyle.test.chapter4formatting.rule4852classannotations;
 
+/**
+ * Test class for checking the rule 4852.
+ */
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java:5]: can we avoid such javadocs ?
this is already in name of fodler and package

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java [hunk 6]
```diff
@@ -1,35 +1,44 @@
...
 public class InputClassAnnotations {
-    public @interface SomeAnnotation1 { }
+  /**
+   * Custom annotation 1.
+   */
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java:9]: lets skip such comments.
Let minimize content that is not related to testing of chapter.

**Review by romani — CHANGES_REQUESTED** [review]
last minors:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java (hunk 6 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java:9]: We require javadocs for interfaces, classes, etc otherwise google config will give warnings

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java (hunk 5 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4852classannotations/InputClassAnnotations.java:5]: We're checking against the whole google config so as of it's rules, we need javadoc for classes, interfaces, etc

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 3 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: done

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 4 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: done

**Review by romani — APPROVED** [review]
ok to merge.

@rnveach , please share your concerns and think if we can improve in separate PR on them.
This PR is "blocking" migration to whole configs, that will be significant effort, but it will NOT touch Abstract classe, so we can do fixes in parallel.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 7]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
         return getModuleConfigsByIds(CONFIGURATION, moduleIds);
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:144]: No public methods except for tests. If something can be made `private`, then it should be that. Otherwise it should be `protected`.

I wouldn't name methods `Google` while in a `Google` abstract class. It sounds like code which calls `toString` on a `String`. Make sure to do this for others and update the javadocs.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 8]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+    }
+
+    /**
+     * Gets the specified children module(s) from the config.
+     *
+     * @param module the children module to get. It can be either a module name or module id.
+     * @return list of children modules.
+     * @throws CheckstyleException if there is a problem for getting the module.
+     */
+    private static List<Configuration> getConfigChildren(String module)
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: This looks like it should be in `AbstractItModuleTestSupport`. I am not seeing anything that says "Google".

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 9]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
+        final Configuration[] children = CONFIGURATION.getChildren();
+        final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
+
+        for (Configuration child : children) {
+            root.addChild(child);
+        }
+
+        verifyWithItConfig(root, filePath);
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: > verifyWithItConfig method in this abstract class, which will take in the entire config.

Why can't we replace `root` parameter with `CONFIGURATION`? The code above looks like it is not "Google" specific.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 10]
```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+                    treeWalkerConfig.addChild(child);
+                }
+            }
+        }
+
+        if (treeWalkerConfig.getChildren().length > 0) {
+            googleConfig.addChild(treeWalkerConfig);
+        }
+
+        verifyWithItConfig(googleConfig, filePath);
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:188]: Same as other `verifyWithItConfig` comment.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (romani) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:144]: as we extend abstract classes, we inherit from most basic methods without "Google" in name, so in final Test class it might be confusing what method to use as it has all methods available for use. Imaginary: verifyWithWholeGoogleConfig , verifyWithItConfig, verifyWithConfig, verifyWithModules

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 8 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: It is used inside verifyWithGoogleConfigParser() :

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L169-L172

It can't be moved to IT abstract class

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:144]: Following up Roman's reply... Currently we're able to access the `verifyWithItConfig` in test files. I couldn't find a way to make it not accessible there. It would be helpful if you can suggest something about this. Thanks for pointing this out

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:144]: > Currently we're able to access the verifyWithItConfig in test files. I couldn't find a way to make it not accessible there. It would be helpful if you can suggest something about this. 

There is no way to prohibit this in Java. We would have to make the method private but then `AbstractGoogle` wouldn't be able to use it. We are stuck for now.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 8 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: Can't `verifyWithGoogleConfigParser` still access it when we switch the method to `protected`? I am still not seeing an issue.

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: Yes it will be accessible. I kept it public because all other `verifyWithxxxxx` methods in `AbstractModuleTestSupport` are also public. 

I'll change this.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 9 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: we cannot directly pass the `CONFIGURATION` to the `verifyWithItConfig()` because `verifyWithItConfig()` uses [getActualViolationsForFile()](https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L481) under the hood and this method uses `createChecker()` inside it:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L484-L487

The `createChecker()` method requires the top level module name to be `root` otherwise it gives error. 

That's why we're getting the whole configuration, changing the top level module name and then passing it to the `verifyWithItConfig()`:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L144-L153


This is also the same reason behind when using the `verifyWithGoogleConfigParser()` method, we create the config with the `root` as top level module:

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L163-L165

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: > The createChecker() method requires the top level module name to be root otherwise it gives error.

This was resolved with the git diff I provided in discord.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 8 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:198]: I am going to drop this move for now.

I still think `AbstractItConfig` is the way to go, but it is more work than we need right now. It will have to be revisted when Sun is updated.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:144]: > I wouldn't name methods Google while in a Google abstract class. 

I still think a lot of "Google" words in this abstract should be removed. Method names and local variables.

Truthfully the only "Google" things in this class appear to be `CONFIGURATION` and `XML_NAME` and neither is named google either. It is because it is Google's configuration and Google's XML. The name is trimmed because the abstract class is Google's.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 10 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:188]: This is also dropped for the same reason, but also this whole method should be removed eventually when we switch everything to the whole config.

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:188]: @Zopsss Where is issue to eventually go back to convert other tests? This should be added to the end there and a comment `until http://....` should be added to this method.

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:185]: https://github.com/checkstyle/checkstyle/issues/14937

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:185]: added `until` message

https://github.com/checkstyle/checkstyle/blob/1b71a042246823ec7fdc3a752575c900c2b2aa14/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java#L459-L467

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 9 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: done added your changes

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: done. removed `google` from every method and javadoc. Changed verify methods names to 

`verifyWithWholeConfig()`
`verifyWithConfigParser()` as discussed in discord

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 9 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: > This is also the same reason behind when using the verifyWithGoogleConfigParser() method, we create the config with the root as top level module:

I have still not changed the code for `verifyWithGoogleConfigParser`, I could not find a way to create config which will have same Checker properties as our google config and I guess it is okay to keep this code as we're not checking against the whole config file, we're checking against only some specific modules

File: src/main/resources/google_checks.xml [hunk 11]
```diff
@@ -21,7 +21,7 @@
 
   <property name="charset" value="UTF-8"/>
 
-  <property name="severity" value="warning"/>
+  <property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
```
Comment (romani) [inline src/main/resources/google_checks.xml:24]: this is not misc update anymore,
I created https://github.com/checkstyle/checkstyle/issues/15053 please move it out to separate PR and we will merge quicker.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 10 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:185]: @Zopsss `until` was placed in wrong location. `until` is basically reminding us the mentioned method is to be removed. `verifyWithItConfig` is the way we will be proceeding now (unless something comes up), so this method should stay.

The note should be on `verifyWithConfigParser` in `AbstractGoogle`. We are only keeping it until all tests move over to full config testing. Once we do, it can be removed completely.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 9 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: As I mentioned in other comment, `verifyWithGoogleConfigParser` should be completely removed eventually, so it is ok that it doesn't do everything since I see it as the old/bad form of testing.

>  I could not find a way to create config which will have same Checker properties as our google config

Just to provide you clarity, what you are missing in the code is calling `CONFIGURATION.getProperties()`. You call get children which only returns the children modules. This is why the original code for `verifyWithWholeConfig` did not pull in the `Checker` properties either.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 8 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:196]: @Zopsss This doesn't look completely done.

> final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:196]: This one is resolved now.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: @Zopsss This doesn't look completely done. I see one left.

> final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 9 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: I am resolving this. If there is any further questions or clarifications needed, please reach out in discord.

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: > As I mentioned in other comment, verifyWithGoogleConfigParser should be completely removed eventually, so it is ok that it doesn't do everything since I see it as the old/bad form of testing.

I think we should keep this method for very large input files. For now lets keep this `until` message, we can discuss this later in discord

> Just to provide you clarity, what you are missing in the code is calling CONFIGURATION.getProperties(). You call get children which only returns the children modules. This is why the original code for verifyWithWholeConfig did not pull in the Checker properties either.

yep I found the way after I commented

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 10 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:186]: done

File: src/main/resources/google_checks.xml (hunk 11 above)
Comment (rnveach) [reply src/main/resources/google_checks.xml:24]: Documentation also needs to be added for this new property. Similar to https://checkstyle.org/google_style.html#Google_Suppressions .

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: @Zopsss 
> No public methods except for tests. If something can be made private, then it should be that. Otherwise it should be protected.

This is not done either. Let me know if there is some conflict.

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: > @Zopsss This doesn't look completely done. I see one left.
> 
> > final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

In this method, we are only loading some specific modules so we should create a config which contains only these modules and their properties. If we create the config whose root will be google_checks.xml's Checker module then we will be copying its properties to the new generated config. So should we really change this and use original Checker module of google_checks.xml as the top level module?

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: I was only commenting to remove `google` from the variable name `googleConfig`. I wasn't trying to change any logic.

File: src/main/resources/google_checks.xml (hunk 11 above)
Comment (Zopsss) [reply src/main/resources/google_checks.xml:24]: made PR on the linked issue, pls check it out

**Comment by rnveach** [comment]
Blocked from merging by https://github.com/checkstyle/checkstyle/pull/15072 , but code updates and reviews can still be done.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 7 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: oh, sorry I misunderstood. Changed the variable name now to `rootConfig`

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java]: > @Zopsss
> 
> > No public methods except for tests. If something can be made private, then it should be that. Otherwise it should be protected.
> 
> This is not done either. Let me know if there is some conflict.
> 
> `verifyWithConfigParser` and `verifyWithWholeConfig` are both `public`.

done. made them protected

**Comment by romani** [comment]
@rnveach , please review this PR as if it does not have update in google_config.xml (moved to separate PR).
if no other comments, please approve PR and I will make sure that https://github.com/checkstyle/checkstyle/pull/15072 is merged first and this PR is rebased.

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java [hunk 12]
```diff
@@ -110,4 +139,71 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
...
+     * Performs verification of the file with the given file path against the whole config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    protected void verifyWithWholeConfig(String filePath) throws Exception {
+        verifyWithItConfig(CONFIGURATION, filePath);
+    }
+
+    // until https://github.com/checkstyle/checkstyle/issues/14937
```
Comment (rdiachenko) [inline src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: Assuming this method will be removed upon migration completion, I'm fine to keep its logic as it is now.

**Review by rdiachenko — APPROVED** [review]
lgtm

File: src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java (hunk 12 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: This method might stay, we will use it for complicated input files for which following all googles rules would be difficult.

Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: This would still go back to why don't we go against the entire config which started us on this path. I understand not everything can be done in the short time of GSoC, but even complicated input files should be converted eventually.

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java:152]: Yess, In GSoC time period I'm not sure if I can convert all tests to use whole config as I have other issues to work on too. But after GSoC, I can complete it.

**Comment by rnveach** [comment]
@Zopsss Please rebase.

**Review by romani — APPROVED** [review]
ok to merge

**Review by rnveach — APPROVED** [review]
